invisible_cities/cities/detsim.py

```diff
@@ -123,6 +124,8 @@ def detsim(*, files_in, file_out, event_range, print_mod, compression,
            detector_db, run_number, s1_lighttable, s2_lighttable, sipm_psf,
            buffer_params, physics_params, rate):
 
+    buffer_params["max_time"] = check_max_time(buffer_params["max_time"], buffer_params["length"])
```

Comment:
We should be more cautious here: `buffer_params` is a dictionary, hence a mutable object. Now you are modifying `buffer_params` inside the function, as it is the case for `physics_params`. In order to avoid unexpected results, I would also duplicate `buffer_params`.

Reply:
Right, in this specific case it is better to do `deepcopy`